### ISSUE TITLE
Multi Task Learning, Include weighted sampling for multiple column

### ISSUE THREAD

**Issue by hamzafar** [horovod/horovod#3761]
**Is your feature request related to a problem? Please describe.**
Dealing with Imbalance class is always been a challenge and for multi-task learning it becomes more challenging. As in multi-task learning we have multiple target label while in simple classification we have one target label. To resolve this, engineers normally adopt for weighted sampling. The technique puts weights on samples (row) based on their class frequency.

In simple Keras script we can perform weighted sampling for multi task learning with the following code:

```
n_samples = x_train.shape[0]
w1 = np.random.rand(n_samples,)
w2 = np.random.rand(n_samples,)

model.fit(x_train, (y_train_1, y_train_2), epochs=10, 
          batch_size=2048, sample_weight={'target_col_1': w1, 'target_col_2': w2})
```

We can use weighted sample technique `sample_weight_col` in `hvd kears estimator` however, this parameter takes only one column. 

**Describe the solution you'd like**
In the horovod api documentation, it is specified that **sample_weight_col – Optional column indicating the weight of each sample.**
Here I would like pass _**list/dictionary of weight columns**_ that can map to different target column like keras example above.
